[export] handle Dim.lower = 0, 1 for ep.run_decompositions()

pianpwk opened the pull request:
Summary:
With pre-dispatch export and ep.run_decompositions(), range constraints are updated through looking at ShapeEnv.var_to_range. However the lower bounds on these may be incorrect - analysis on un-specialized symbols are done with lower bounds of 2, which mismatch with user-specified bounds (may be 0, 1).

This updates `_get_updated_range_constraints()` to use the old range constraints if possible.

Test Plan: Existing pre-dispatch/dynamic shapes test case.

Differential Revision: D55899872

tugsbayasgalan commented on test/export/test_export.py:
```diff
@@ -899,7 +899,6 @@ def forward(self, x, y, z):
         ):
             _ = export(foo, inputs, dynamic_shapes=((dx, 9), (dy, 4), (3, 3)))
 
-    @testing.expectedFailurePreDispatchRunDecomp  # T183703911
```
Can you rebase on https://github.com/pytorch/pytorch/pull/123326 ? I did some refactoring on that diff

pianpwk committed:
[export] handle Dim.lower = 0, 1 for ep.run_decompositions() (#123602)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/123602

With pre-dispatch export and ep.run_decompositions(), range constraints are updated through looking at ShapeEnv.var_to_range. However the lower bounds on these may be incorrect - analysis on un-specialized symbols are done with lower bounds of 2, which mismatch with user-specified bounds (may be 0, 1).

This updates `_get_updated_range_constraints()` to use the old range constraints if possible.

Test Plan: Existing pre-dispatch/dynamic shapes test case.

Reviewed By: tugsbayasgalan

Differential Revision: D55899872

izaitsevfb commented:
@pytorchbot merge -f 'Landed internally'
